clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetchRequestManagerTest.java

```diff
@@ -3386,6 +3394,71 @@ public void testWhenFetchResponseReturnsALeaderShipChangeErrorAndNewLeaderInform
         assertTrue(subscriptions.isFetchable(tp1));
     }
 
+    @Test
+    public void testPollWithoutCreateFetchRequests() {
+        buildFetcher();
+
+        assignFromUser(singleton(tp0));
+        subscriptions.seek(tp0, 0);
```

Comment:
Prior to this change, all that was needed to start fetching was a valid set of assigned partitions. Post-change, that is insufficient; no fetch requests will be issued if not explicitly requested.

If I removed the assignment, the test looks very anemic:

```java
    @Test
    public void testPollWithoutCreateFetchRequests() {
        buildFetcher();
        assertEquals(0, sendFetches(false));
    }
```

I'm OK to remove that test since the other tests cover that basic test anyway.

Thoughts?

Reply:
thinking more I say it's better to keep them even if they are setting up a condition that it's truly not used in the test. The reason is really that if ever there's a regression and we wrongfully send a "ghost" request in the sendFetches(false), we don't want to miss catching it just because some other conditions are missing.